### PR TITLE
lxd/db/cluster: Extend list of entities

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

**Comment by stgraber** [comment]
Looks good.
One thing I thought of just now, can we make `TypeContainer` use fixed integers for each of them?
That will allow for me to backport this to stable-4.0 with some gaps in the middle for the APIs which don't exist in 4.0 without causing all the identifiers to change when someone upgrades (which would be a problem as the warnings will persist in the database).

**Comment by monstermunchkin** [comment]
>  One thing I thought of just now, can we make TypeContainer use fixed integers for each of them?

@stgraber better? :)

File: lxd/db/cluster/constants.go [hunk 1]
```diff
@@ -9,29 +9,68 @@ import (
...
 var EntityTypes = map[string]int{}
 
 // EntityURIs associates an entity code to its URI pattern.
 var EntityURIs = map[int]string{
-	TypeContainer: "/" + version.APIVersion + "/containers/%s?project=%s",
-	TypeImage:     "/" + version.APIVersion + "/images/%s",
-	TypeProfile:   "/" + version.APIVersion + "/profiles/%s?project=%s",
-	TypeProject:   "/" + version.APIVersion + "/projects/%s",
+	TypeContainer:             "/" + version.APIVersion + "/containers/%s?project=%s",
+	TypeImage:                 "/" + version.APIVersion + "/images/%s",
```
Comment (stgraber) [inline lxd/db/cluster/constants.go:58]: Not sure why this one isn't using ?project=%s.
Can you make a separate commit to correct it and track down where that format string is used to see if we need to pass it the project name at that spot?

Comment (monstermunchkin) [reply lxd/db/cluster/constants.go:58]: @stgraber `EntityURIs[TypeImage]` was used for creating [`projects_used_by_ref`](https://github.com/lxc/lxd/blob/master/lxd/db/cluster/update.go#L2355-L2368), but the latest change to `projects_used_by_ref` [hard-codes the URIs](https://github.com/lxc/lxd/blob/master/lxd/db/cluster/update.go#L118-L158).

**Comment by stgraber** [comment]
You're going to need a new DB update that deletes that view and recreates it.
Along with a `make update-schema` to update the generated schema.

**Comment by monstermunchkin** [comment]
> You're going to need a new DB update that deletes that view and recreates it.
> Along with a make update-schema to update the generated schema.

Why? The `projects_used_by_ref` view is created correctly in `updateFromV45`.

**Comment by stgraber** [comment]
Ah, okay, I missed that. Then we don't need the commit fixing V11 as the later update already corrects it.

**Comment by monstermunchkin** [comment]
> Ah, okay, I missed that. Then we don't need the commit fixing V11 as the later update already corrects it.

@stgraber But won't LXD complain about the URI string having one too many `%s`? V11 doesn't pass the project name, but the URI has a placeholder for it.

**Comment by stgraber** [comment]
@monstermunchkin all database updates are applied on LXD startup, so we'll never run a query using current LXD code with the V11 schema, instead on startup LXD will update to the latest schema which you said looks good and that will happen before we issue any query.

**Comment by monstermunchkin** [comment]
@stgraber I removed the last commit. We're good to go.
